spec/models/application_record_spec.rb

```diff
@@ -67,6 +67,21 @@
     end
   end
 
+  describe 'data_rows' do
+    let :call do
+      @record.data_rows
+    end
+    before :each do
+      @record = create :application_record,
+                       data: [['a question', 'an answer']]
```

Comment:
Not at the time it was written, I suspect. Both add_data_types and add_response_data are only used in lib tasks--used when migrating data from the initial app state.